test/libgit2-online.jl

```diff
@@ -21,28 +21,29 @@ mktempdir() do dir
                 finalize(repo)
             end
         #end
-        try
+
         #@testset "with incorrect url" begin
-            repo_path = joinpath(dir, "Example2")
-            # credential are required because github try authenticate on unknown repo
-            cred = LibGit2.UserPasswordCredentials("","") # empty credentials cause authentication error
-            LibGit2.clone(https_prefix*repo_url*randstring(10), repo_path, payload=Nullable(cred))
-            error("unexpected")
+            try
+                repo_path = joinpath(dir, "Example2")
+                # credentials are required because github tries authenticate on unknown repo
+                cred = LibGit2.UserPasswordCredentials("","") # empty credentials cause authentication error
+                LibGit2.clone(https_prefix*repo_url*randstring(10), repo_path, payload=Nullable(cred))
+                error("unexpected")
+            catch ex
+                @test isa(ex, LibGit2.Error.GitError)
+                @test ex.code == LibGit2.Error.EAUTH
+            end
         #end
-        catch ex
-            @test isa(ex, LibGit2.Error.GitError)
-            @test ex.code == LibGit2.Error.EAUTH
-        end
 
-        try
-            #@testset "with 'ssh' protocol (by default is not supported)" begin
+        #@testset "with 'ssh' protocol (by default is not supported)" begin
+            try
                 repo_path = joinpath(dir, "Example3")
                 @test_throws LibGit2.Error.GitError LibGit2.clone(ssh_prefix*repo_url, repo_path)
```

Comment:
We deal with this test when ssh will be available. Yes, features can be checked to see which protocols are supported, but that is not part of this PR. The rework of the url handling, similar to #13879, should include it.